azavea/raster-foundry
# Connect AOI front-end to API

## Overview

This PR connects most of the AOI front-end to the API.

**A user can now:**
- create an AOI project from the 'Create Project' modal
- adjust various parameters of an AOI project (update frequency, start-time)
- approve, reject, or do nothing with pending scenes

**This PR also cleans up some parts of the UI:**
- adds new header style to color correction portions of UI
- removes the scene detail overlay in favor of the scene detail modal
- adds event propagation halting to buttons within scene items so that click 'Remove from project' does not delete the scene from the project AND show the scene detail modal
- adds a scene count to the project scene list

**Things this PR doesn't do but you might find yourself thinking it should:**
1) allow drawing your AOI area (this is required to create an AOI object)
2) allow persisting of the AOI filters...because of 1
3) allow creating the actual AOI that would be referenced by your AOI project...because of 1. 

That being said, for points 2 and 3 once the AOI area can be drawn in, all that would need to happen is to uncomment some code that is indicated by a `TODO` and these things would be sent to the API.

4) set a project as needing approval for scene addition - we will be making this the default for now, but it needs to be handled on the backend (see testing instructions for getting around this)

### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] Swagger specification updated, if necessary~
~- [ ] Symlinks from new migrations present or corrected for any new migrations~

## Testing Instructions

(Prepare yourself)

 * Create a new project but this time select 'Area of Interest Project'!
 * Select 'Add public imagery'
 * You'll be taken to the add imagery page of the editor, as expected, but click back (using the arrow in the sidebar pane) which will take you to the project scene list.
 * You shouldn't have any scenes in your project at this point, and the scene list should have a message indicating that
 * Click back once more to the main editor area. There should be an 'AOI PROJECT' tag in the header bar along with a link to 'AOI Parameters'. Click into the parameters.
 * You can adjust both the update frequency and the start date. Remember what values you set them to. After you click 'Start Monitoring AOI', you should be taken back to the main editor.
 * Go back into the AOI parameters and ensure the params are the same as the ones you set
 * Test that the filter pane opens as expected (filter values will not be persisted until we can draw and create valid AOI objects)
 * Go back to the main editor page and then proceed to the imagery browser to add some 5-10 scenes to your project
 * If you go back a step to the project scenes list, you should see the scenes you just added (as usual)
 * Copy your project id and then open a psql shell (`./scripts/psql`) and then update the `scenes_to_projects` table so that scenes within the project (`project_id` is the column) have `accepted=false`.
 * In the UI, if you reload (and you're on the project scenes list page), the scenes should disappear, but you should see a banner indicating that you have scenes to approve.
 * Click the 'Review' button and then start accepting or rejecting scenes and see that it all works as you'd expect
 * Try some other combinations of things and make sure nothing breaks

Closes #1667 
Closes #1668 
Closes #1748 
Closes #1726
